src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+  };
+
+  /**
+   * Instantiates a NeynarV1APIClient
+   *
+   * Note: A Wallet must be provided if the API client is to mint new AuthTokens
+   */
+  constructor(
+    apiKey: string,
+    {
+      logger = silentLogger,
+      axiosInstance,
+    }: { logger?: Logger; axiosInstance?: AxiosInstance } = {}
+  ) {
+    this.logger = logger;
+
+    if (apiKey === undefined) {
+      throw new Error(
+        "Attempt to use an authenticated API method without first providing an api key"
+      );
+    }
+
+    if (axiosInstance === undefined) {
+      axiosInstance = axios.create();
+    }
+    axiosInstance.defaults.decompress = true;
+    axiosInstance.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        if (NeynarAPIClient.isApiErrorResponse(error)) {
+          const apiErrors = error.response.data;
+          this.logger.warn(`API errors: ${JSON.stringify(apiErrors)}`);
+        }
+        throw error;
+      }
+    );
+
+    const v1Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v1",
+      apiKey: apiKey,
+    });
+    const v2Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v2",
+      apiKey: apiKey,
+    });
+    this.apis = {
+      v1Cast: new CastApiV1(v1Config, undefined, axiosInstance),
+      user: new UserApi(v1Config, undefined, axiosInstance),
+      verification: new VerificationApi(v1Config, undefined, axiosInstance),
+      notifications: new NotificationsApi(v1Config, undefined, axiosInstance),
+      reactions: new ReactionsApi(v1Config, undefined, axiosInstance),
+      v1Follows: new FollowsApiV1(v1Config, undefined, axiosInstance),
+      v2Cast: new CastApiV2(v2Config, undefined, axiosInstance),
+      signer: new SignerApi(v2Config, undefined, axiosInstance),
+      reaction: new ReactionApi(v2Config, undefined, axiosInstance),
+      v2Follow: new FollowApiV2(v2Config, undefined, axiosInstance),
+    };
+  }
+
+  /**
+   * Utility for parsing errors returned by the Merkle API server. Returns true
+   * if the given error is caused by an error response from the server, and
+   * narrows the type of `error` accordingly.
+   */
+  public static isApiErrorResponse(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    error: any
+  ): error is WithRequired<AxiosError<ErrorRes>, "response"> {
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
+  }
+
+  /**
+   * Creates a Signer. See [Neynar documentation](https://docs.neynar.com/reference/create-signer)
+   * for more details.
+   */
+  public async createSigner(): Promise<Signer> {
+    const response = await this.apis.signer.createSigner();
+    return response.data;
+  }
+
+  /**
+   * Fetches an existing Signer. See [Neynar documentation](https://docs.neynar.com/reference/get-signer)
+   * for more details.
+   *
+   */
+  public async fetchSigner(signerUuid: string): Promise<Signer | undefined> {
```

Comment:
In writing the `hub-rest` client I revisited the question of whether we should return `undefined` or `null` when a requested thing isn't found.

Which do you think? I can be convinced either way.

I see that the Neynar v1 swagger spec is using the `nullable: true` option in at least one place. If Neynar is going with `nulls` everywhere then maybe we might want to as well for consistency (I'm not sure if they are, I just saw the one `nullable` instance)

Reply:
They do use null throughout the spec, I'm going to follow that pattern and replace undefined with nulls